addons/mail/models/mail_mail.py

```diff
@@ -123,32 +124,65 @@ def process_email_queue(self, ids=None):
         return res
 
     @api.multi
-    def _postprocess_sent_message(self, mail_sent=True):
+    def _postprocess_sent_message(self, success_email_list, failure=False, email=None):
```

Comment:
where does the filter_notification_ids come from in this example? one more param coming from _notify?  

To notify success, we need to know the failed message 
success_notif.mapped('mail_message_id')._notify_send_success() is not correct since it is possible to have a message with a failed notif and a success notif. We realy need to have something like all_message - failed message, all_message - failed_notif.mapped('mail_message_id')

We also need to be sure that there isn't any notification linked to this message but not linked to any mail in this postprocess in cancel or sent state. (we resend a mail to some partner but not all, some notifications will stay in exception state, we can't remove the failure)

                    failed_messages = failed.mapped('mail_message_id')
                    if failed_messages:
                        self._notify_failure_update(failed_messages, False)
                    if notify_success:
                        all_messages = notifications.mapped('mail_message_id')
                        #we want to be sure that message has no notification with failure left before sending a remove error notif
                        messages_to_notify = [message for message in (all_messages - failed_messages) if not [notif for notif in message.notification_ids if notif.email_status not in ('sent', 'canceled')]]
                        self._notify_failure_update(messages_to_notify, True)

To simplify this, we could simply send a generic message with all notifications and let js do the work. We dont notify add_failure or remove_failure but failure update.